packages/services/src/services/VideoPlayer/VideoPlayer.js

```diff
@@ -201,6 +203,13 @@ class VideoPlayerAPI {
         readyCallback: function(playerId) {
           var kdp = document.getElementById(playerId);
 
+          var container = document.querySelector(
+            `.${prefix}--modal-close, ${prefix}-modal-close-button`
+          );
+          container.addEventListener('click', function() {
+            kdp.sendNotification('doStop');
+          });
+
```

Comment:
Instead of baking in modal close handler here (given the code here is a service API library), we can make this API return `kdp` so that the caller can control the video play. In this way, our React code and our Web Components code can handle modal close event by themselves. So we can start from something like below?:

```diff
diff --git a/packages/services/src/services/VideoPlayer/VideoPlayer.js b/packages/services/src/services/VideoPlayer/VideoPlayer.js
index 0f8b3581..036d3a74 100644
--- a/packages/services/src/services/VideoPlayer/VideoPlayer.js
+++ b/packages/services/src/services/VideoPlayer/VideoPlayer.js
@@ -184,6 +184,7 @@ class VideoPlayerAPI {
   static async embedVideo(videoId, targetId, autoPlay) {
     const fireEvent = this.fireEvent;
     return await this.checkScript().then(() => {
+      const promiseKWidget = new Promise(resolve => {
         root.kWidget.embed({
           targetId: targetId,
           wid: '_' + _partnerId,
@@ -210,9 +211,17 @@ class VideoPlayerAPI {
             kdp.addJsListener('playerPlayEnd', function() {
               fireEvent({ playerState: 3, kdp, videoId });
             });
+
+            resolve(kdp);
           },
         });
       });
+      return {
+        kWidget() {
+          return promiseKWidget;
+        },
+      };
+    });
   }
 
   /**
```